js/factsheet-archive.js

```diff
@@ -3,5 +3,43 @@
 		$( ".degree-list" ).on( "click", ".degree-row-multiple .degree-name", function( el ) {
 			$( el.target ).closest( ".degree-row-wrapper" ).toggleClass( "degree-row-open" );
 		} );
+
+		$( "#degree-search-input" ).on( "change blur keyup", function() {
+			doSearch( $( "#degree-search-input" ).val() );
+		} );
+
+		$( ".degree-list .pagination a" ).on( "click", function() {
+			$( ".degree-list .pagination a.active" ).removeClass( "active" );
+			$( this ).addClass( "active" );
+		} );
 	} );
+
+	function scrollToElement( elem ) {
+		$( "html, body" ).animate( {
+			scrollTop: $( elem ).offset().top - 60
+		}, 800 );
+	}
+
+	function doSearch() {
+		var terms = $( ".lettergroup li .degree-name" ).map( function() { return this.innerText; } ).get();
+
+		$( "#degree-search-input" ).autocomplete( {
+			source: terms,
+			open: function() {
+				$( "body" ).addClass( "searching" );
+			},
+			close: function() { $( "body" ).removeClass( "searching" ); },
+			select: function( event, ui ) {
+				$( "body" ).removeClass( "searching" );
+				var text = ui.item.value.replace( /(\r\n|\n|\r)/gm, "" );
+				var degrees = $( ".degree-name a:contains('" + text + "')" );
```

Comment:
I think the `a` should be dropped from this selector, for cases of multiple degrees with the same name. The degree name in those cases is wrapped in a `span`, causing this to throw `Uncaught TypeError: Cannot read property 'top' of undefined`.